### PR TITLE
[FLINK-27013][hive] Hive dialect supports is_distinct_from

### PR DESCRIPTION
**Pull request by luoyuxia**
## What is the purpose of the change
To support `is_distinct_from` with Hive dialect.

## Brief change log
Add the implementation for `is_distinct_from` in [ExprCodeGenerator.scala](https://github.com/apache/flink/pull/19315/files#diff-2a1762af3f6bd3f984cdf27261f217384278f302437c632f32f35a60ffa1e624).

## Verifying this change
Added test involving with is_distinct_from in [HiveDialectQueryITCase.java](https://github.com/apache/flink/pull/19315/files#diff-fb5fc4106e27e27f9968e59b643003444fd40af58aa2f147ac301009b53a47ca)#testDistinctFrom

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:  no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? N/A

### CODE REVIEW

File: flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala [hunk 1]
```diff
@@ -445,14 +445,24 @@ object ScalarOperatorGens {
...
+
   def generateIsNotDistinctFrom(
       ctx: CodeGeneratorContext,
       left: GeneratedExpression,
       right: GeneratedExpression)
     : GeneratedExpression = {
     generateOr(
       generateAnd(generateIsNull(left), generateIsNull(right)),
-      generateEquals(ctx, left, right))
+      generateIsTrue(generateEquals(ctx, left, right)))
```
Comment (luoyuxia) [inline flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala:465]: generateEquals may produce UNKNOWN, so we should change it to generateIsTrue(generateEquals).

**Comment by luoyuxia** [comment]
@flinkbot run azure

**Comment by lsyldliu** [comment]
@luoyuxia  Thanks for your contribution, LGTM

**Review by lsyldliu — APPROVED** [review]

**Comment by luoyuxia** [comment]
@flinkbot run azure

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**